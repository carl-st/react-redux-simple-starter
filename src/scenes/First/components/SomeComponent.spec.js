import React from 'react';
import { shallow } from 'enzyme';
import sinon from 'sinon';
import { SomeComponent } from ".";

describe('<SomeComponent />', () => {
    const props = {
        image: 'http://example.com/',
        subtitle:'Test subtitle',
        title: 'Test title',
        onClick: () => {}
    };
    
    it('renders correctly with provided props', () => {
        const wrapper = shallow(<SomeComponent {...props} />);

        expect(wrapper).toMatchSnapshot();
    });

    it('should call function passed as [onClick] prop', () => {
        const onClickSpy = sinon.spy();
        const wrapper = shallow(<SomeComponent {...props} onClick={onClickSpy} />);

        wrapper.find('div').first().simulate('click');

        expect(onClickSpy.called).toBe(true);
    });


});
