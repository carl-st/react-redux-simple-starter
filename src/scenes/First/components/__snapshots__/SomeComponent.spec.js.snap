// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SomeComponent /> renders correctly with provided props 1`] = `
<React.Fragment>
  <div
    onClick={[Function]}
  />
</React.Fragment>
`;
